clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -218,6 +225,10 @@ public void wakeup() {
 
     }
 
+    public Set<TopicPartition> paused() {
+        return paused;
```

Comment:
See, here you should evaluate what is the semantic that ``paused`` should have. If we want to return a **snapshot** of the paused ``TopicPartition`` then it's better to do:

```
return Collections.unmodifiableSet(new HashSet<>(paused));
```

If we want to return a **dynamic** view of the ``paused`` then it's better to use:

```
return Collections.unmodifiableSet(paused);
```

In either case, we should return an unmodifiable view of the set, because it's not very nice to expose a field directly to callers as above.